### PR TITLE
Add Travis and Codecov

### LINKED ISSUE #6: Changes to boilerplate pre-development

### PR DESCRIPTION
**Pull request by jonseitz**
A relatively small PR that required a bunch of extra commits for testing purposes. BUT NOW WE HAVE GREEN CHECKMARKS AGAIN!

Along the way I also had to make some changes to our webpackfile because some stage in the pipeline was causing a memory leak that was failing every build on Travis. Now our chunks should be all in a row.  

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to change)

## Checklist:

- [x] I have run `eslint` on the code
- [x] I have added JSDoc for all of my code (where applicable)
- [x] I have added tests to cover my changes.

## Priority:
- [x] Normal 
- [ ] High 

## Related Issues:

Fixes seas-computing/course-planner#6

### CODE REVIEW

File: webpackfile.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const webpack = require('webpack');
 const HtmlWebpackPlugin = require('html-webpack-plugin');
 const HtmlWebpackRootPlugin = require('html-webpack-root-plugin');
 const TSConfigPathsPlugin = require('tsconfig-paths-webpack-plugin');
-
+const TerserPlugin = require('terser-webpack-plugin');
 /**
  * NestJs uses a custom wrapper around require() for allows it to show a
```
Comment (rmainwork) [inline webpackfile.js]: Not part of this change-set, but I just noticed this grammar error in the comment here :upside_down_face:

File: webpackfile.js [hunk 2]
```diff
@@ -26,73 +26,95 @@ const nestBlacklist = [
...
+};
+
 const client = {
   name: 'client',
-  mode: 'production',
+  mode,
   entry: ['./src/client/index.ts'],
   output: {
-    path: resolve(__dirname, 'dist/static'),
+    path: resolve(__dirname, 'build/static'),
```
Comment (rmainwork) [inline webpackfile.js:78]: What's the difference between `build/` and `dist/`?

**Review by rmainwork — APPROVED** [review]
I'm approving this, this looks good to me. I do have a couple of questions, but generally - great work!

File: webpackfile.js (hunk 1 above)
Comment (jonseitz) [reply webpackfile.js]: ah good point. I'll go ahead and fix that in 1a69c85

File: webpackfile.js (hunk 2 above)
Comment (jonseitz) [reply webpackfile.js:78]: None, really. I think the typescript default is `dist/`, but most of our other configurations use `build/`, particularly in our `Dockerfile` where [it copies the `build` dir from the `builder` stage][dockerfile] into the production image. As you would expect, trying to copy from a non-existent directory doesn't work. 

[dockerfile]: https://github.com/seas-computing/react-boilerplate/blob/64c36ae69a61333a4a2ca6ec5376c33165935f9f/Dockerfile#L24

Comment (rmainwork) [reply webpackfile.js:78]: Fair enough. I just wondered if there was anything different going on here :)
